### PR TITLE
Move `statespace` module out of `filtsmooth` into top level

### LINKED ISSUE #282: (Statespace) models
Currently, transition rules and the like are in `filtsmooth.statespace`. I'd like to pull them out and make them their own thing. In this context, I think it might be interesting to rename it from `statespace` to something like `models` (as in: *probabilistic models*) because all those transitions can essentially be used as likelihoods for all kinds of things. 

Maybe at some point, we will be able to remove the filtsmooth dependency from `diffeq` and only have a `statespace` (`models` resp.) dependency.

Thoughts on the renaming, @nathanaelbosch @JonathanWenger ? I guess no one would complain about moving it up...

### PR DESCRIPTION
**Pull request by schmidtjonathan**
Closes #282 .

### CODE REVIEW

File: docs/source/public_api/filtsmooth.rst [hunk 1]
```diff
@@ -12,17 +12,3 @@ observations into account.
...
-
-
-probnum.filtsmooth.statespace
-=============================
-
-Probabilistic State Space Models.
-
-This package implements continuous-discrete and discrete-discrete state
-space models, which are the basis for Bayesian filtering and smoothing,
-but also probabilistic ODE solvers.
```
Comment (pnkraemer) [inline docs/source/public_api/filtsmooth.rst:24]: What do you think of the following changes of wording?

implements **components of** continuous-discrete and ...

the basis for BFAS, but also **for** probabilistic ODE solvers

File: src/probnum/filtsmooth/gaussfiltsmooth/extendedkalman.py [hunk 2]
```diff
@@ -31,7 +31,7 @@ def forward_realization(
         compute_gain=False,
         _diffusion=1.0,
         _linearise_at=None,
-    ) -> (pnrv.Normal, typing.Dict):
+    ) -> typing.Tuple[pnrv.Normal, typing.Dict]:
```
Comment (pnkraemer) [inline src/probnum/filtsmooth/gaussfiltsmooth/extendedkalman.py:34]: good catch. Thanks.

**Review by pnkraemer — APPROVED** [review]
Looks good already, no complaints on my side!  :)

Thanks for doing this!

**Comment by schmidtjonathan** [comment]
@pnkraemer I had a rebase conflict after updating the branch, so it would be nice, if you could check if everything is still alright, before I merge :)
Thanks!

File: src/probnum/diffeq/odefiltsmooth/odefiltsmooth.py [hunk 3]
```diff
@@ -362,31 +362,31 @@ def _string2matern(ivp, which_prior, **kwargs):
...
+        return pnss.Matern(
             3,
             ivp.dimension,
             lengthscale,
             forward_implementation="sqrt",
             backward_implementation="sqrt",
         )
     elif which_prior == "matern92":
-        return pnfs.statespace.Matern(
+        return pnss.Matern(
```
Comment (pnkraemer) [inline src/probnum/diffeq/odefiltsmooth/odefiltsmooth.py:389]: I know this is annoying now, and that it is not part of your PR, but this line (and the related ones above) are not covered by tests (for some reason, codecov thinks this is your fault) which leads the the diff coverage being <90% (which is why you wont be able to merge I guess). What do you think about adding a quick test for these settings in ``test_odefiltsmooth.TestLotkaVolterraOtherPriors``? Or, alternatively, remove these options (as they've never been used anyway -- in the future, we will have to consider how to handle the order of the prior in a separate issue; see #272.

**Review by pnkraemer — COMMENTED** [review]
overall, looks good. There is this diff coverage thing, which is why the CI is failing. See my comment below.

File: src/probnum/diffeq/odefiltsmooth/odefiltsmooth.py (hunk 3 above)
Comment (schmidtjonathan) [reply src/probnum/diffeq/odefiltsmooth/odefiltsmooth.py:389]: I will add a test for that and then re-request you as a reviewer 👍

**Pull request merged**